### PR TITLE
Batter way to write binarize() function.

### PR DESCRIPTION
**Pull request by hi-sushanta**
The elegance of this image binarization code lies in its simplicity. With just a single line, it leverages the PIL.Image.point() method to threshold the image into black and white. Free of complexity and verbose syntax, this solution gets straight to the point - converting pixel values to binary based on a set threshold. 

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [x] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?
@sayakpaul @patrickvonplaten @yiyixuxu

### CODE REVIEW

File: src/diffusers/image_processor.py [hunk 1]
```diff
@@ -400,8 +400,7 @@ def binarize(self, image: PIL.Image.Image) -> PIL.Image.Image:
             `PIL.Image.Image`:
                 The binarized image. Values less than 0.5 are set to 0, values greater than 0.5 are set to 1.
         """
-        image[image < 0.5] = 0
-        image[image >= 0.5] = 1
+        image.point(lambda p: 0 if p < 0.5 else 1, "1")
```
Comment (sayakpaul) [inline src/diffusers/image_processor.py:403]: The code above is more readable and `image.point()` is not extremely widely used unlike the other PIL methods that we use. 

@yiyixuxu WDYT?

Comment (hi-sushanta) [reply src/diffusers/image_processor.py]: Ok, that's fine.

File: src/diffusers/image_processor.py [hunk 2]
```diff
@@ -33,14 +33,7 @@
...
 
-PipelineDepthInput = Union[
-    PIL.Image.Image,
-    np.ndarray,
-    torch.FloatTensor,
-    List[PIL.Image.Image],
-    List[np.ndarray],
-    List[torch.FloatTensor],
-]
+PipelineDepthInput = PipelineImageInput
```
Comment (patrickvonplaten) [inline src/diffusers/image_processor.py:36]: Nice!

File: src/diffusers/image_processor.py [hunk 3]
```diff
@@ -153,27 +146,21 @@ def convert_to_rgb(image: PIL.Image.Image) -> PIL.Image.Image:
         """
         Converts a PIL image to RGB format.
         """
-        image = image.convert("RGB")
-
-        return image
+        return image.convert("RGB")
```
Comment (patrickvonplaten) [inline src/diffusers/image_processor.py:149]: This makes the code harder to debug with `pdb` or `ipdb` - do we really need this change?

**Review by yiyixuxu — COMMENTED** [review]
can we revert all the changes except https://github.com/huggingface/diffusers/pull/6394/files#r1439464834

thanks!

File: src/diffusers/image_processor.py (hunk 2 above)
Comment (hi-sushanta) [reply src/diffusers/image_processor.py:36]: Thank you.

File: src/diffusers/image_processor.py (hunk 3 above)
Comment (hi-sushanta) [reply src/diffusers/image_processor.py:149]: You're right! Your suggestions were invaluable, and I've gladly returned to the original version. Thank you for guiding me towards the best outcome.

**Comment by hi-sushanta** [comment]
I recognized the value in those two crucial lines, so I carefully preserved them while reverting to other changes.

**Review by yiyixuxu — APPROVED** [review]

**Review by sayakpaul — APPROVED** [review]

**Pull request merged**